## Block 1

In [35]:
# Libraries being imported
import numpy as np
import pandas as pd

## Block 2

In [56]:
# Loading the data
data = pd.read_csv('problem_set_1_data.csv')

# Rename the columns
data.columns = ['quiz_total','attended','final_grade','final_exam','name']

# Getting head and tail of data frame plus summary
head = data.head()
print('------------ Head ------------')
print('Head of data = ', head)

tail = data.tail()
print('------------ Tail ------------')
print(tail)


summary = data.describe()
print('------------ Summary ------------')
print (summary)

------------ Head ------------
Head of data =     quiz_total  attended  final_grade  final_exam               name
0       78.50         1        99.38          46       Linnie Lietz
1       64.25         0        95.78          45      Mitch Mustain
2       76.75         0        97.42          41     Salina Chavera
3       69.00         1        92.50          38  Kimberely Conwell
4       64.00         0        92.50          39          Zack Burk
------------ Tail ------------
     quiz_total  attended  final_grade  final_exam                 name
166        73.5         0        96.25          45           Dana Malta
167        75.5         0        98.13          44         Sixta Heyden
168        41.0         0        69.38          32       Brianne Broome
169        77.0         0       101.25          47  Elouise Weatherholt
170        72.0         0        92.81          38        Addie Maharaj
------------ Summary ------------
       quiz_total    attended  final_grade  fina

There is a total of 171 students that attended this class. The quiz mean was a 69.1, final exam was 39.7 and the final grade was a 91.86 which brings the class to an average of an A-. We learn that mostly everyone did great in the class as the 25 percentile of the class had an 89 average final grade which is a B+. The lowest grade was a 0 but that was probably those who didn't attend class/dropped the class.

## Block 3

In [58]:
# Did attend class
boolean_index0 = data['attended'] != 0
yes_attendance = data[boolean_index0]
y_attendance = yes_attendance.copy()

In [60]:
# Did not attend class
boolean_index1 = data['attended'] != 1
no_attendance = data[boolean_index1]
n_attendance = no_attendance.copy()

In [62]:
# averages of both
yes_summary = y_attendance.describe()
print('------------ Did go to class ------------')
print(yes_summary)

no_summary = n_attendance.describe()
print('------------ Did not go to class ------------')
print(no_summary)

------------ Did go to class ------------
       quiz_total  attended  final_grade  final_exam
count   33.000000      33.0    33.000000   33.000000
mean    72.500000       1.0    95.072121   40.636364
std      5.547029       0.0     4.673980    4.014179
min     54.000000       1.0    85.470000   33.000000
25%     70.000000       1.0    91.250000   38.000000
50%     74.000000       1.0    94.690000   40.000000
75%     77.000000       1.0    99.380000   44.000000
max     78.500000       1.0   101.250000   47.000000
------------ Did not go to class ------------
       quiz_total  attended  final_grade  final_exam
count  138.000000     138.0   138.000000  138.000000
mean    68.326087       0.0    91.086159   39.442029
std     10.057028       0.0    10.174000    5.800853
min      0.000000       0.0     2.500000    0.000000
25%     65.000000       0.0    88.320000   37.000000
50%     71.000000       0.0    92.770000   40.000000
75%     75.375000       0.0    96.210000   43.750000
max     79.

The data shows that 33 people did show up to class while 138 did not. Those who showed up to class had on average a higher quiz total (+4%), slighty higher final exam grade (+1%), and a higher final grade (+4%) than those who didn't. This goes to show that going to class could mean the difference between getting an A- or getting an A+.